File: static/js/workers.js
async function workerSignup() {

    const name = document.getElementById("name").value
    const email = document.getElementById("email").value
    const phone = document.getElementById("phone").value
    const password = document.getElementById("password").value
    const state = document.getElementById("state").value

    const response = await fetch("/api/worker-signup", {

        method: "POST",

        headers: {
            "Content-Type": "application/json"
        },

        body: JSON.stringify({
            name: name,
            email: email,
            phone: phone,
            password: password,
            state: state
        })

    })

    const data = await response.json()

    if (data.status === "success") {

        alert("Signup Successful")

        window.location.href = "/login"

    } else {

        alert(data.message)

    }

}


async function updateWorkerProfile() {

    const email = document.getElementById("email").value
    const address = document.getElementById("address").value
    const qualification = document.getElementById("qualification").value
    const experience = document.getElementById("experience").value
    const skills = document.getElementById("skills").value

    const response = await fetch("/api/worker-profile", {

        method: "POST",

        headers: {
            "Content-Type": "application/json"
        },

        body: JSON.stringify({
            email: email,
            address: address,
            qualification: qualification,
            experience: experience,
            skills: skills
        })

    })

    const data = await response.json()

    if (data.status === "success") {

        alert("Profile Updated")

    } else {

        alert("Error updating profile")

    }

}


async function loadWorkers() {

    const response = await fetch("/api/workers")

    const workers = await response.json()

    const container = document.getElementById("workersContainer")

    container.innerHTML = ""

    workers.forEach(worker => {

        const card = document.createElement("div")

        card.className = "worker-card"

        card.innerHTML = `
            <h3>${worker.name}</h3>
            <p>Skills: ${worker.skills || "N/A"}</p>
            <p>Experience: ${worker.experience || "N/A"}</p>
            <p>Email: ${worker.email}</p>
            <p>Phone: ${worker.phone}</p>
        `

        container.appendChild(card)

    })

}


async function searchWorkers() {

    const skill = document.getElementById("skillSearch").value

    const response = await fetch(`/api/search-workers/${skill}`)

    const workers = await response.json()

    const container = document.getElementById("workersContainer")

    container.innerHTML = ""

    workers.forEach(worker => {

        const card = document.createElement("div")

        card.className = "worker-card"

        card.innerHTML = `
            <h3>${worker.name}</h3>
            <p>Skills: ${worker.skills}</p>
            <p>Experience: ${worker.experience}</p>
            <p>Email: ${worker.email}</p>
            <p>Phone: ${worker.phone}</p>
        `

        container.appendChild(card)

    })

}

File: backend/workers.py
from fastapi import APIRouter
from pydantic import BaseModel
from database import workers_collection

router = APIRouter()


class WorkerSignup(BaseModel):
    name: str
    email: str
    phone: str
    password: str
    state: str


class WorkerProfile(BaseModel):
    email: str
    address: str
    qualification: str
    experience: str
    skills: str


@router.post("/api/worker-signup")
def worker_signup(data: WorkerSignup):

    existing = workers_collection.find_one({"email": data.email})

    if existing:
        return {"status": "error", "message": "Email already exists"}

    workers_collection.insert_one(data.dict())

    return {"status": "success", "message": "Worker registered successfully"}


@router.post("/api/worker-profile")
def worker_profile(data: WorkerProfile):

    workers_collection.update_one(
        {"email": data.email},
        {"$set": {
            "address": data.address,
            "qualification": data.qualification,
            "experience": data.experience,
            "skills": data.skills
        }}
    )

    return {"status": "success", "message": "Profile updated"}


@router.get("/api/workers")
def get_workers():

    workers = list(workers_collection.find({}, {"_id": 0}))

    return workers


@router.get("/api/search-workers/{skill}")
def search_workers(skill: str):

    workers = list(workers_collection.find(
        {"skills": {"$regex": skill, "$options": "i"}},
        {"_id": 0}
    ))

    return workers
